lib/phoenix_live_view/lifecycle.ex

```diff
@@ -0,0 +1,217 @@
+defmodule Phoenix.LiveView.Hook do
+  @moduledoc false
+
+  if Version.match?(System.version(), ">= 1.8.0") do
+    @derive {Inspect, only: [:id, :stage]}
+  end
+
+  defstruct id: nil, stage: nil, function: nil
+
+  @type t :: %__MODULE__{
+          id: term(),
+          stage: :handle_event | :handle_info | :handle_params | :mount,
+          function: function()
+        }
+
+  def new!(id, stage, fun) when is_atom(stage) and is_function(fun) do
+    %__MODULE__{id: id, stage: stage, function: fun}
```

Comment:
To be honest, do we need this to be a struct/separate module? Keeping it as maps inside lifecycle should be enough, no? And given lifecycle is already a struct, if we want to implement inspect, implementing it for the lifecycle is enough?

Reply:
Sounds good!